pkg/router/admin.go

```diff
@@ -2,118 +2,252 @@ package router
+			logger.Errorw("failed to get coupons", "error", err)
+			return echo.NewHTTPError(http.StatusInternalServerError)
+		}
+		return c.JSON(http.StatusOK, result)
 	}
 }
 
-// @Summary Admin Get Coupon Detail
-// @Description Get coupon details.
-// @Tags Admin, Coupon, Shop
-// @Produce json
-// @Param id path int true "Coupon ID"
-// @Success 200
-// @Failure 401
-// @Router /admin/coupon/{id} [get]
+// @Summary		Admin Get Coupon Detail
+// @Description	Get coupon details.
+// @Tags			Admin, Coupon, Shop
+// @Produce		json
+// @Param			id	path		int	true	"Coupon ID"
+// @Success		200	{object}	db.GetCouponDetailRow
+// @Failure		400	{object}	echo.HTTPError
+// @Failure		404	{object}	echo.HTTPError
+// @Failure		500	{string}	echo.HTTPError
+// @Router			/admin/coupon/{id} [get]
 func adminGetCouponDetail(pg *db.DB, logger *zap.SugaredLogger) echo.HandlerFunc {
 	return func(c echo.Context) error {
-		return c.NoContent(http.StatusOK)
+		id := c.Param("id")
+		idInt, err := strconv.ParseInt(id, 10, 32)
+		if err != nil {
+			logger.Errorw("failed to parse id", "error", err)
+			return echo.NewHTTPError(http.StatusBadRequest)
+		}
+		idInt32 := int32(idInt)
+		if exist, err := pg.Queries.CouponExists(c.Request().Context(), idInt32); err != nil {
+			logger.Errorw("failed to check coupon exist", "error", err)
+			return echo.NewHTTPError(http.StatusInternalServerError)
+		} else if !exist {
+			logger.Infow("coupon not found", "id", id)
+			return echo.NewHTTPError(http.StatusNotFound, "Coupon not found")
+		}
+		if result, err := pg.Queries.GetCouponDetail(c.Request().Context(), idInt32); err != nil {
+			logger.Errorw("failed to get coupon detail", "error", err)
+			return echo.NewHTTPError(http.StatusInternalServerError)
+		} else {
+			return c.JSON(http.StatusOK, result)
+		}
 	}
 }
 
-// @Summary Admin Add Coupon
-// @Description Add global coupon.
-// @Tags Admin, Coupon
-// @Accept json
-// @Produce json
-// @Success 200
-// @Failure 401
-// @Router /admin/coupon [post]
+// @Summary		Admin Add Coupon
+// @Description	Add global coupon.
+// @Tags			Admin, Coupon
+// @Accept			json
+// @Produce		json
+// @Param			coupon	body		db.AddCouponParams	true	"Coupon"
+// @Success		200		{object}	db.AddCouponRow
+// @Failure		400		{object}	echo.HTTPError
+// @Failure		500		{object}	echo.HTTPError
+// @Router			/admin/coupon [post]
 func adminAddCoupon(pg *db.DB, logger *zap.SugaredLogger) echo.HandlerFunc {
 	return func(c echo.Context) error {
-		return c.NoContent(http.StatusOK)
+		var coupon db.AddCouponParams
+		if err := c.Bind(&coupon); err != nil {
+			logger.Errorw("failed to bind coupon", "error", err)
+			return echo.NewHTTPError(http.StatusBadRequest)
+		}
+		coupon.Scope = "global"
+		result, err := pg.Queries.AddCoupon(c.Request().Context(), coupon)
+		if err != nil {
+			logger.Errorw("failed to add coupon", "error", err)
+			return echo.NewHTTPError(http.StatusInternalServerError)
+		}
+		return c.JSON(http.StatusOK, result)
 	}
 }
```

Comment:
insert coupon without shopid

Reply:
since it is admin, it don't have shop_id